webcompat/api/endpoints.py

```diff
@@ -52,10 +52,13 @@ def edit_issue(number):
     """XHR endpoint to push back edits to GitHub for a single issue.
 
     - It only allows change of state and change of milestones.
-    - It is always proxied to allow any logged in user
-      to be able to edit issues.
+    - It's not proxied, so only users with write access are
+      able to edit issues.
       Format: {'milestone': 2, 'state': 'open'}
     """
+    if not g.user:
+        abort(403)
+
```

Comment:
Added this check here because `api_request` conveniently sends `proxy_request` if `g.user` doesn't exist :D

https://github.com/webcompat/webcompat.com/blob/main/webcompat/helpers.py#L474

Reply:
We probably need to work out a bit the architecture and double check what is still using `proxy_request` and how. 

And then simplify things.